src/Reflection/DataTransferObjectProperty.php

```diff
@@ -82,27 +82,17 @@ private function resolveCaster(): ?Caster
         /** @var \Spatie\DataTransferObject\Attributes\CastWith $attribute */
         $attribute = $attributes[0]->newInstance();
 
+        $typeNames = array_map(fn ($type) => $type->getName(), $this->extractTypes());
+
         return new $attribute->casterClass(
-            $this->reflectionProperty->getType()?->getName(),
+            implode('|', $typeNames),
```

Comment:
How would this work without making changes to the `CastWith::__construct` attribute? It still expects a single class name. I understand the issue but I don't see how this PR fixes it.

Reply:
`CastWith` attribute doesn't need any changes AFAIK, but two supplied casters do. I've replaced "|"-delimited string with type name array and made needed changes to the casters.